resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -320,50 +322,83 @@ private[spark] class KubernetesClusterSchedulerBackend(
     override def eventReceived(action: Action, pod: Pod): Unit = {
       val podName = pod.getMetadata.getName
       val podIP = pod.getStatus.getPodIP
-
+      val podPhase = pod.getStatus.getPhase
       action match {
-        case Action.MODIFIED if (pod.getStatus.getPhase == "Running"
+        case Action.MODIFIED if (podPhase == "Running"
             && pod.getMetadata.getDeletionTimestamp == null) =>
           val clusterNodeName = pod.getSpec.getNodeName
           logInfo(s"Executor pod $podName ready, launched at $clusterNodeName as IP $podIP.")
           executorPodsByIPs.put(podIP, pod)
 
-        case Action.DELETED | Action.ERROR =>
+        case Action.MODIFIED if (podPhase == "Init:Error" || podPhase == "Init:CrashLoopBackoff")
```

Comment:
Maybe have a separate set for all the error states we want to check. Having one place would make this easier to change in future.

Reply:
Are `Init:Error` and `Init:CrashLoopBackoff ` valid `PodPhase` values? I don't see them in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2215. It appears to me that those are printed form of` PodStatus`. Or the fabric8 client we use return those?

Reply:
The Kubernetes client doesn't use any enumerations from the underlying API, it only takes the raw strings in the response body. So if the response gives us those values, we should be fine.

Reply:
I took a look at the client code, and it appears to me that `getPhase` simply returns the value of json property `phase` of `PodStatus`. Have you seen `Init:Error` as the return value in practice?

Reply:
Seen it using `kubectl`, haven't tried hitting the API directly. When we start seeing these kinds of errors come up again, we can check our pod YAML.

Reply:
Yes, you will see if with `kubectl describe`. But I doubt you will see it with the client library. This whole PR makes little sense if those values are never seen. I think you need to check both the pod phase (if it is `Failed`) and the `Initialized` condition is `true` to determine if the pod failed on any init-container.

Reply:
The termination reason is also a good source of info. kubectl looks at a set of these and turns it into what you see in the describe output - so, [similar logic](https://github.com/kubernetes/kubernetes/blob/6b94e872c63eeea2ed4fdc510c008e4ff9713953/pkg/printers/internalversion/printers.go#L547-L573) could be exercised.